readme.md

```diff
@@ -535,6 +536,19 @@ const randIter = rand()
 randIter.next() // Each execution gives a random value, expression is evaluated on need.
 ```
 
+## Thunk
+
+A thunk is another word for a function used to inject an additional calculation into another function (or simply a function that is returned by another function). Thunks are primarily used to delay a calculation until its result is needed, or to insert operations at the beginning or end of the other function.
```

Comment:
Can get bit more better I feel, what say @jethrolarson ?

Reply:
Ping @kr5hn4

Reply:
Ok, how about this @hemanth 

A thunk is a function/subroutine with zero parameters used to delay a calculation until its result is needed.

For Eg. Say you have to add two numbers `7+4` in your program, rather than writing `7+4` in your program you create a function with zero arguments like
```
function myThunk() {
  return 7+4;
}
```

So by creating a function your program will not immediately calculate the result of `7+4` but rather give you a function which you can call later when the result of `7+4` is actually needed.

Reply:
Sounds better, thanks! 

We could use `const getBigNumber = () => factorial(1000000);` see the comment below.

Reply:
@hemanth I have deleted the forked repo, could you please make the changes?